--- gnulib/doc/INSTALL	2017-02-23 15:26:15.296354123 +0000
+++ INSTALL	2017-12-08 15:38:22.819949859 +0000
@@ -1,3 +1,44 @@
+Building and installing libenchant
+**********************************
+
+For standard GNU autotools installation instructions, see below.  There
+follow some enchant-specific notes.
+
+The usual GNU autotools package instructions apply, except that Enchant must
+be built with --enable-relocatable for the tests to run successfully.
+
+
+Building from git
+=================
+
+See HACKING.
+
+
+Parallel installation
+=====================
+
+Enchant installs in versioned directories, so that different major
+versions can be installed in parallel.
+
+
+Relocation
+==========
+
+If Enchant is installed in a fixed location, and on Windows and Linux
+systems, no further configuration should be needed to make it relocatable;
+that is, work when copied from its original installation directory to
+another location.
+
+Otherwise, and in code intended to be portable to other systems, your code
+should call enchant_set_prefix_dir (see enchant.h) before any other Enchant
+function.
+
+For more details of how the relocation support works, including how you can
+use it in your own program, see doc/relocatable.texi and
+gnulib/doc/relocatable-maint.texi.
+
+
+
 Installation Instructions
 *************************
 
