--- gnulib/m4/warnings.m4	2017-02-23 15:26:15.536364509 +0000
+++ gl/m4/warnings.m4	2017-04-04 22:27:00.204131193 +0100
@@ -74,6 +74,20 @@
          [AC_SUBST([WARN_CFLAGS])])dnl
 ])
 
-# Local Variables:
-# mode: autoconf
-# End:
+# gl_CXX_WARN_ADD(OPTION, [VARIABLE = WARN_CFLAGS],
+#             [PROGRAM = AC_LANG_PROGRAM()])
+# ---------------------------------------------
+# Adds parameter to WARN_CXXFLAGS if the compiler supports it when
+# compiling PROGRAM.  For example, gl_CXX_WARN_ADD([-Wparentheses]).
+#
+# If VARIABLE is a variable name, AC_SUBST it.
+AC_DEFUN([gl_CXX_WARN_ADD],
+[AC_REQUIRE([gl_UNKNOWN_WARNINGS_ARE_ERRORS])
+gl_COMPILER_OPTION_IF([$1],
+  [gl_AS_VAR_APPEND(m4_if([$2], [], [[WARN_CXXFLAGS]], [[$2]]), [" $1"])],
+  [],
+  [$3])
+m4_ifval([$2],
+         [AS_LITERAL_IF([$2], [AC_SUBST([$2])])],
+         [AC_SUBST([WARN_CXXFLAGS])])dnl
+])
